tradeexecutor/visual/technical_indicator.py

```diff
@@ -33,15 +36,73 @@ def overlay_all_technical_indicators(
         Crop range
     """
 
+    # get starting row for indicators
+    cur_row = _get_initial_row(volume_bar_mode)
+
     # https://plotly.com/python/graphing-multiple-chart-types/
-    for plot_id, plot in visualisation.plots.items():
+    for plot in visualisation.plots.values():
+        
         trace = visualise_technical_indicator(
             plot,
             start_at,
             end_at,
         )
-        if trace is not None:
-            fig.add_trace(trace)
+        
+        # must have something to draw for plot
+        if trace is None:
+            raise ValueError(f"Unknown plot kind: {plot.plot_kind}")
+        
+        # add trace to plot
+        if plot.kind == PlotKind.technical_indicator_detached:
+            cur_row += 1
+            fig.add_trace(trace, row=cur_row, col=1)
+        elif plot.kind == PlotKind.technical_indicator_on_price:
+            fig.add_trace(trace, row=1, col=1)
+        else:
+            raise ValueError(f"Unknown plot kind: {plot.plot_kind}")
+
+        # add horizontal line if needs be
+        _add_hline(fig, start_at, end_at, cur_row, plot)
+
+        
+
+def _get_initial_row(volume_bar_mode):
+    """Get first row for plot based on volume bar mode."""
+    if volume_bar_mode in {VolumeBarMode.hidden, VolumeBarMode.overlay}:
+        cur_row = 1
+    elif volume_bar_mode == VolumeBarMode.separate:
+        cur_row = 2
+    else:
+        raise ValueError("Unknown volume bar mode: %s" % volume_bar_mode)
+    return cur_row
+
+def _add_hline(fig, start_at, end_at, cur_row, plot: Plot):
+    """Add horizontal line to plot if needed."""
+    if line_val := plot.horizontal_line:
```

Comment:
It's a bad practice to do conditional functions.

It is better to check this at the call site and then only call the function if the conditions are sufficient